android/src/main/kotlin/com/signify/hue/flutterreactiveble/channelhandlers/ScanDevicesHandler.kt

```diff
@@ -46,15 +46,19 @@ class ScanDevicesHandler(private val bleClient: com.signify.hue.flutterreactiveb
     }
 
     fun stopDeviceScan() {
-        scanForDevicesDisposable.dispose()
-        scanParameters = null
+        if (this::scanForDevicesDisposable.isInitialized) scanForDevicesDisposable.let {
```

Comment:
Why `scanForDevicesDisposable.let {}` is needed here? In `this::scanForDevicesDisposable.isInitialized` you've already assumed `scanForDevicesDisposable` is non-null (if that is relevant).

Reply:
It is a bit of sugar to put it in a let block not per se needed but also no harm. Btw I did .let and not ?.let so indeed I do not check on nullability (Detekt would also complain in that case)